### DISCUSSION TITLE
added char hexconst[] = {'0','1','2','3','4','5','6','7','8','9','A', 'B', 'C', 'D', 'E', 'F',...'}; in rxvmintp.c

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#222] [Q&A]
For an easier hex conversion, I added the variable hexconst[]. I am not sure if the location is the correct one:

```
...
int run(int num_modules, module *program, int argc, char *argv[],
        int debug_mode) {
...
struct avl_tree_node *exposed_proc_tree = 0;
    struct avl_tree_node *exposed_reg_tree = 0;
    value *g_reg;
    char hexconst[] = {'0','1','2','3','4','5','6','7','8','9','A', 'B', 'C', 'D', 'E', 'F','a', 'b', 'c', 'd', 'e', 'f'};
#ifndef NUTF8
    int codepoint;
#endif
...
```

**Comment by adesutherland** [comment]
That's not the best place because it will be added to each exposed procedure definition ...

However have you considered https://en.cppreference.com/w/c/string/byte/isxdigit?

**Reply by Peter-Jacob** [reply]
this is cool, but I need it the other way round:   for '0'x I must add the char '1', for '15'x it must be 'F' and so on. It's the backward translation into a hex number (used byc2X). I'd appreciate a clever solution, if there isn't one, tell me where to move the char hexconst[ definition.

**Reply by adesutherland** [reply]
OK - will suggest somewhere - late afternoon, i am afraid ... meetings

**Reply by Peter-Jacob** [reply]
for you:   [poor boy](https://www.youtube.com/watch?v=ZwAe6CCqwUc)
for me:    [I am free](https://www.youtube.com/watch?v=uRD_gIoVOmY)

**Comment by Peter-Jacob** [comment]
@adesutherland Do you have a suggestion for me?

**Reply by adesutherland** [reply]
Sorry sorry! And I had misinterpreted where you wanted to put it!

So - yes where you suggest is perfect ...
